src/uct/cuda/cuda_copy/cuda_copy_md.c

```diff
@@ -30,12 +30,32 @@ static ucs_config_field_t uct_cuda_copy_md_config_table[] = {
 
 static ucs_status_t uct_cuda_copy_md_query(uct_md_h md, uct_md_attr_t *md_attr)
 {
+    ucs_status_t status;
+    int          active;
+    size_t       total;
+    CUdevice     dev;
+
+    UCT_CUDADRV_CTX_ACTIVE(active);
+    if (!active) {
+        return UCS_ERR_NO_DEVICE;
```

Comment:
Prefer to not return error but to use `total = 0` when context is not found. If md_query is called during ucp_init (as part of MPI_Init) and an error is returned, it may end up leading to cuda transports being unusable later on (supposing that cuda context gets set after ucp_init but before ucp_tag_send_nb)